agents/holo_synth/synth_agent.py

```diff
@@ -0,0 +1,232 @@
+            self.lo_id = synth3.get_LOs()
+
+            synth3.set_RF_output(0, 1, self.lo_id)  # LO ID, On=1, USB connection ID
+            synth3.set_RF_output(1, 1, self.lo_id)  # LO ID, On=1, USB connection ID
+
+            # data = {"timestamp": time.time(), "block_name": "synth_lo", "data": {}}
+
+            # data["data"]["F1_status"] = 1
+            # data["data"]["F2_status"] = 1
+
+            # self.agent.publish_to_feed("synth_lo", data)
+            # session.data.update(data["data"])
+
+        # This part is for the record and to allow future calls to proceed,
+        # so does not require the lock
+        self.initialized = True
+
+        return True, "Synth Initialized."
+
+    @ocs_agent.param("offset", type=float, default=0, check=lambda x: 0 <= x <= 1000)
+    @ocs_agent.param("freq1", type=float, default=0, check=lambda x: 0 <= x <= 1000)
+    def set_frequencies(self, session, params):
+        """set_frequencies(freq1=0, offset=0)
+
+        **Task** - A task to set the frequencies of the synthesizers.
+
+        Parameters:
+            freq1 (float): Frequency of holography measuremnt [GHz].
+            offset (float): Frequency offset of holography measurement [MHz].
+
+        Examples:
+            Example for calling in a client::
+
+                from ocs.ocs_client import OCSClient
+                client = OCSClient("synth_lo")
+                client.set_frequencies(freq1=210, offset=10)
+        """
+        f1 = params.get("freq1", 0)
+        f_offset = params.get("offset", 0)
+
+        F_1 = int(f1 * self.ghz_to_mhz / self.N_MULT)  # Convert GHz -> MHz for synthesizers
+
+        with self.lock.acquire_timeout(timeout=3, job="set_frqeuencies") as acquired:
+            if not acquired:
+                self.log.warn(
+                    f"Could not set position because lock held by {self.lock.job}"
+                )
+                return False, "Could not acquire lock"
+
+            synth3.set_f(0, F_1, self.lo_id)
+            synth3.set_f(1, F_1 + f_offset, self.lo_id)
+
+            data = {"timestamp": time.time(), "block_name": "synth_lo", "data": {}}
+
+            data["data"]["F1"] = F_1
+            data["data"]["F2"] = F_1 + f_offset
+
+            self.agent.publish_to_feed("synth_lo", data)
+            session.data.update(data["data"])
+
+        return True, "Frequencies Updated"
+
+    # This function is not finished, need to figure out how to read out frequency from USB connection.
+    # def read_frequencies(self, session, params=None):
+
+    #     with self.lock.acquire_timeout(timeout=3, job="read_frqeuencies") as acquired:
+    #         if not acquired:
+    #             self.log.warn(
+    #                 f"Could not set position because lock held by {self.lock.job}"
+    #             )
+    #             return False, "Could not acquire lock"
+
+    #     return True, "Frequencies Updated"
+
+    @ocs_agent.param("lo_id_n", type=int, default=0, choices=[0, 1])
```

Comment:
"lo_id_n" is inconsistent with the documented 'lo_id' parameter, and with the use of `params.get('lo_id')`. Make consistent.